src/data/markdown/docs/05 Examples/01 Examples/19 functional testing.md

```diff
@@ -44,134 +44,1528 @@ This test goes through several steps. It creates a new user account, authenticat
+    var myObj = { val: 1 }
+      , addTwo = function () { myObj.val += 2; };
+
+    expect(addTwo).to.increase(myObj, 'val').by(3, 'nooo why fail??');
+    expect(addTwo, 'nooo why fail??').to.increase(myObj, 'val').by(3);
+  });
+
+  describe('docs example 193', () => {
+    expect({ a: 1 }).to.be.extensible;
+  });
+
+  describe('docs example 194', () => {
+    var nonExtensibleObject = Object.preventExtensions({})
+      , sealedObject = Object.seal({})
+      , frozenObject = Object.freeze({});
+
+    expect(nonExtensibleObject).to.not.be.extensible;
+    expect(sealedObject).to.not.be.extensible;
+    expect(frozenObject).to.not.be.extensible;
+    expect(1).to.not.be.extensible;
+  });
+
+  describe('docs example 195', () => {
+    expect(1, 'nooo why fail??').to.be.extensible;
+  });
+
+  describe('docs example 196', () => {
+    var sealedObject = Object.seal({});
+    var frozenObject = Object.freeze({});
+
+    expect(sealedObject).to.be.sealed;
+    expect(frozenObject).to.be.sealed;
+    expect(1).to.be.sealed;
+  });
+
+  describe('docs example 197', () => {
+    expect({ a: 1 }).to.not.be.sealed;
+  });
+
+  describe('docs example 198', () => {
+    expect({ a: 1 }, 'nooo why fail??').to.be.sealed;
+  });
+
+  describe('docs example 199', () => {
+    var frozenObject = Object.freeze({});
+
+    expect(frozenObject).to.be.frozen;
+    expect(1).to.be.frozen;
+  });
+
+  describe('docs example 200', () => {
+    expect({ a: 1 }).to.not.be.frozen;
+  });
+
+  describe('docs example 201', () => {
+    expect({ a: 1 }, 'nooo why fail??').to.be.frozen;
+  });
+
+  describe('docs example 202', () => {
+    expect(1).to.be.finite;
+  });
+
+  describe('docs example 203', () => {
+    expect('foo').to.be.a('string'); // Recommended
+    expect('foo').to.not.be.finite; // Not recommended
+  });
+
+  describe('docs example 204', () => {
+    expect(NaN).to.be.NaN; // Recommended
+    expect(NaN).to.not.be.finite; // Not recommended
+  });
+
+  describe('docs example 205', () => {
+    expect(Infinity).to.equal(Infinity); // Recommended
+    expect(Infinity).to.not.be.finite; // Not recommended
+  });
+
+  describe('docs example 206', () => {
+    expect(-Infinity).to.equal(-Infinity); // Recommended
+    expect(-Infinity).to.not.be.finite; // Not recommended
+  });
+
+  describe('docs example 207', () => {
+    expect('foo', 'nooo why fail??').to.be.finite;
+  });
+
+  describe('docs example 208', () => {
+    expect.fail();
+    expect.fail("custom error message");
+    expect.fail(1, 2);
+    expect.fail(1, 2, "custom error message");
+    expect.fail(1, 2, "custom error message", ">");
+    expect.fail(1, 2, undefined, ">");
+  });
+
+  // describe('docs example 209', () => {
+  //   should.fail();
+  //   should.fail("custom error message");
+  //   should.fail(1, 2);
+  //   should.fail(1, 2, "custom error message");
+  //   should.fail(1, 2, "custom error message", ">");
+  //   should.fail(1, 2, undefined, ">");
+
+  // });
```

Comment:
The example is commented out. Is that expected?

Should we add a quick example for `should`?

The following gave the expected result:

```javascript
import http from 'k6/http';
import { describe, expect, chai } from 'https://jslib.k6.io/k6chaijs/4.3.4.0/index.js';

export default function () {
  describe('Get Public Crocodiles',   () => {
    const res = http.get('https://test-api.k6.io/public/crocodiles');
    chai.should();
    res.should.have.validJsonBody();
});
}
```

It is actually the first style listed on the [ChaiJS](https://www.chaijs.com/) website (although I'm not sure if that means it is widely used - I don't really like this style TBH).